src/herbie.rkt

```diff
@@ -13,7 +13,11 @@
  ["improve.rkt" (run-improve)])
 
 (define (string->thread-count th)
-  (match th ["no" #f] ["yes" (max (- (processor-count) 1) 1)] [_ (string->number th)]))
+  (match th
+    ["no" #f]
+    ["yes" (max (- (processor-count) 1) 1)]
+    [(? positive-integer?) (string->number th)]
```

Comment:
I don't think this makes sense? `th` is a string, so it will never satisfy `positive-integer?`? I get that you want to catch the case of negative (or zero or fractional) thread count, but I think you need a separate `match` or `cond`.